azure/azure_kv.go

```diff
@@ -213,3 +233,17 @@ func init() {
 		panic(err.Error())
 	}
 }
+
+func getAzureClientOptions(secretConfig map[string]interface{}) azcore.ClientOptions {
+	opts := azcore.ClientOptions{}
+	cloudEnv := getAzureKVParams(secretConfig, AzureEnviornment)
+	if strings.EqualFold(cloudEnv, "AzureUSGovernment") {
+		opts.Cloud = cloud.AzureGovernment
+	} else if strings.EqualFold(cloudEnv, "AzureChinaCloud") {
+		opts.Cloud = cloud.AzureChina
+	} else {
+		opts.Cloud = cloud.AzurePublic
+	}
```

Comment:
Do we need to do this? Can we not pass a user-defined string to the libopenstorage/secrets lib here so that Rook doesn't need to do custom processing?

Reply:
The current version in the master is using [this](https://github.com/libopenstorage/secrets/blob/5f4b25ceede180e8d67577288bba4223af658eb1/azure/azure_kv.go#L68-L71). So if we don't do this it might be a breaking change for users. 

I'll check again if we really need this or not.

Reply:
Maybe we can add this logic or these `"AzureUSGovernment"`/`"AzureChinaCloud"` strings to the libopenstorage library. But in general, I am concerned about hard coding these values. 

It's entirely possible and reasonable that Azure might add other clouds like `"AzureEuropeCloud"`, `"AzureRussiaCloud"`, or any other reasonably big use-case where data security needs to be different from other clouds, and if we force any unrecognized option to use the public cloud, those users might be broken. It seems best to me just to pass in the value given to Rook into the lib without trying to validate it.

Reply:
I was using `AzurePublicCloud` for any unrecognized option. That's wrong.  I've updated to use only `AzurePublicCloud` only if the user has set it or if user does not provide an option.

Reply:
>>  It seems best to me just to pass in the value given to Rook into the lib without trying to validate it.

I think this would result in breaking changes to the users already using this library. Because they are only expected to provide the `AZURE_ENVIRONMENT` and then libopenshift/secrets finds out the exact environment based on the `AZURE_ENVIRONMENT`